File: IT/2/Multithreading/Java/Multithreading2.java
/* file Multithreading2.java */

/* la classe MyThread rappresenta un Thread personalizzato che esegue un blocco di codice */

class MyThread extends Thread {
	@Override
	public void run() {
		System.out.println("Ciao, sono il " + this.getName() + "!");
	}
}

public class Multithreading2 {
	public static void main(String [] args) {
		/* crea due nuovi thread di tipo MyThread */

		MyThread t1 = new MyThread();
		MyThread t2 = new MyThread();

		/* abilita i nuovi thread per l'esecuzione */

		t1.start();
		t2.start();
	}
}

File: IT/2/Multithreading/C++/multithreading2.cpp
/* 
file multithreading2.cpp 
compilare con: g++ -std=c++11 multithreading2.cpp -lpthread
*/

#include <cstdio>
#include <thread>

using std::thread;

void func(int n) {
	printf("Ciao, sono il thread %d!\n", n);
}

int main()
{
	/* crea due thread che eseguono la funzione func() */

	thread t1(func, 1);
	thread t2(func, 2); 

	t1.join(); 
	t2.join(); 
}

